pyccel/ast/internals.py

```diff
@@ -153,6 +157,8 @@ def __init__(self, start, stop, step = None):
             raise TypeError('Slice stop must be Integer or None')
         if step is not None and not (hasattr(step, 'dtype') and isinstance(step.dtype, NativeInteger)):
             raise TypeError('Slice step must be Integer or None')
+        if slice_type is not None and not (hasattr(slice_type, 'dtype') and isinstance(slice_type.dtype, NativeInteger)):
```

Comment:
What happens if `slice_type is None`?
The second half can be written more compactly as
`isinstance(getattr(slice_type, 'dtype', None), NativeInteger))`

But should this not be:
`if slice_type not in (Range, Element):`
?

Reply:
`self.Range, self.Element` to pass